src/sentry/sentry_metrics/consumers/indexer/slicing_router.py

```diff
@@ -0,0 +1,83 @@
+from typing import MutableMapping, Optional
+
+from arroyo import Message, Topic
+from arroyo.backends.kafka import KafkaPayload
+from confluent_kafka import Producer
+from django.conf import settings
+
+from sentry.ingest.partitioning import (
+    Sliceable,
+    is_sliced,
+    map_logical_partition_to_slice,
+    map_org_id_to_logical_partition,
+)
+from sentry.sentry_metrics.consumers.indexer.routing_producer import MessageRoute, MessageRouter
+from sentry.utils import kafka_config
+
+
+class SlicingRouter(MessageRouter[KafkaPayload]):
+    def __init__(
+        self,
+        sliceable: Sliceable,
+        logical_output_topic: str,
+    ) -> None:
+        self.__sliceable = sliceable
+        self.__logical_output_topic = logical_output_topic
+        self.__slice_to_producer: MutableMapping[int, MessageRoute] = {}
+
+        if not is_sliced(self.__sliceable):
+            self.__slice_to_producer[0] = MessageRoute(
+                producer=Producer(
+                    kafka_config.get_kafka_producer_cluster_options(
+                        settings.KAFKA_TOPICS[self.__logical_output_topic]["cluster"]
+                    )
+                ),
+                topic=Topic(self.__logical_output_topic),
```

Comment:
Don't we want to be able to map this to a different physical topic?

Reply:
I think the name `logical_output_topic` is a misnomer. I will fix it. The current state is that output messages get sent to an output topic. In order to allow transitioning from unsliced output to sliced output, we need capability to support both.